lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -192,12 +192,13 @@ def self.escape_quotes(text)
 
   def self.redact_content(content)
     # Note that here we match after an equals to ensure we redact the value being passed, but not the key.
-    # This means a redaction of a string not including '= ' before the string value will not redact.
+    # This means a redaction of a string not including '= ' before the string value will not redact. We
+    # also match everything before the match and backreference it in the replacement string so we don't
+    # lose other key/value pairs on the same line.
     # Every secret unwrapped in this module will unwrap as "'secret' # PuppetSensitive" and, currently,
     # always inside a hash table to be passed along. This means we can (currently) expect the value to
-    # always come after an equals sign.
-    # Note that the line may include a semi-colon and/or a newline character after the sensitive unwrap.
-    content.gsub(/= '.+' # PuppetSensitive;?(\\n)?$/,"= '[REDACTED]'")
+    # always come after an equals sign and only exist once per line.
+    content.gsub(/(.*)= '.+' <# PuppetSensitive #>/,"\\1= '[REDACTED]'")
```

Comment:
Slightly more functional check via @Iristyle that ensures we redact any number of sensitive objects on a line:

```ruby
content.gsub(/= '.+?' <# PuppetSensitive #>/,"= '[REDACTED]'")
```

Reply:
👍